crates/katana/primitives/src/genesis/json.rs

```diff
@@ -312,34 +320,99 @@ impl TryFrom<GenesisJson> for Genesis {
                                 serde_json::from_value(artifact.clone())?;
                             casm.class_hash()?
                         };
-
                         (class_hash, class_hash, None, Arc::new(CompiledContractClass::V0(casm)))
                     }
                 };
 
+                // Add the class name and class hash to the mapping
+                let _ = name.map(|name| {
```

Comment:
We don't want the user to define 2 classes with the same name. Could we add a check to ensure that the names are unique for only ONE class?